ocaml/xapi/xapi_vm_helpers.ml

```diff
@@ -983,11 +983,11 @@ let vm_fresh_genid ~__context ~self =
 
 (** Add to the VM's current operations, call a function and then remove from the
 	current operations. Ensure the allowed_operations are kept up to date. *)
-let with_vm_operation ~__context ~self ~doc ~op ?policy f =
+let with_vm_operation ~__context ~self ~doc ~op ?(force=false) ?policy f =
 	let task_id = Ref.string_of (Context.get_task_id __context) in
 	Helpers.retry_with_global_lock ~__context ~doc ?policy
 		(fun () ->
-			Xapi_vm_lifecycle.assert_operation_valid ~__context ~self ~op;
+			if not force then Xapi_vm_lifecycle.assert_operation_valid ~__context ~self ~op;
```

Comment:
The function `Xapi_vm_lifecycle.assert_operation_valid` does a lot more checks than just the tools check. Do we really want `force` to be able to bypass all of these? I think that we would risk getting into a bad state if we do that.